packages/components/src/presentation-components/tree/controlled/UseTreeReload.ts

```diff
@@ -38,96 +40,97 @@ export function useTreeReload(params: TreeReloadParams & { enable: boolean }) {
 }
 
 function useModelSourceUpdateOnIModelHierarchyUpdate(params: TreeReloadParams & { enable: boolean }): void {
-  const { enable, dataProviderProps, rulesetId, pageSize, modelSource, onReload, renderedItems } = params;
+  const { enable, dataProviderProps, ruleset, pageSize, modelSource, onReload, renderedItems } = params;
 
   useEffect(() => {
-    if (!enable || !rulesetId || !modelSource) {
+    if (!enable || !modelSource) {
       return;
     }
 
     let subscription: Subscription | undefined;
     const removeListener = Presentation.presentation.onIModelHierarchyChanged.addListener((args: IModelHierarchyChangeEventArgs) => {
-      if (args.rulesetId !== rulesetId || args.imodelKey !== dataProviderProps.imodel.key) {
+      if (args.rulesetId !== ruleset || args.imodelKey !== dataProviderProps.imodel.key) {
```

Comment:
Should `args.rulesetId !== ruleset` check instead use `getRulesetId(ruleset)`?

Reply:
Missed this one.